url78eec0e0b6_Orikuro_vic_tgz-pJ8-net_nexon_vindictus_itemcomparer_inport_StaticImportJ8/src/net/nexon/vindictus/itemcomparer/logic/ComboCall.java

```diff
@@ -0,0 +1,350 @@
+
+	private List<Combo> combos = new ArrayList<>();
+
+	private double total_p = 0;
+	private double s_p = 0;
+	private double p_p = 0;
+	private double g_p = 0;
+	private double a_p = 0;
+	private double h_p = 0;
+	private int thresh;
+
+	@Override
+	public List<Combo> call() throws Exception {
+
+		String compname = comp.getClass().getSimpleName();
+
+		switch (compname) {
+		case "AtkComparator":
+			return callAtk();
+		case "DefComparator":
+			return callDef();
+		case "MatkComparator":
+			return callMatk();
+		case "StaDComparator":
+		case "StaAComparator":
+		case "StaMComparator":
+			return callSta();
+		default:
+			return callStandard();
+		}
+	}
+
+	private List<Combo> callStandard() throws Exception {
+		int i = 0;
+		for (Shoes s : shoes) {
+			s_p = s.getTotalPrice();
+			for (Pants p : pants) {
+				p_p = p.getTotalPrice();
+				for (Gloves g : gloves) {
+					g_p = g.getTotalPrice();
+					for (Armor a : armors) {
+						a_p = a.getTotalPrice();
+						for (Helm h : helms) {
+							h_p = h.getTotalPrice();
+							total_p = s_p + p_p + g_p + a_p + h_p;
+
+							if (total_p <= PRICE) {
+								combos.add(new Combo(s, p, g, a, h));
+							}
+							i++;
+							if (i % 1000000 == 0) {
+								System.out.print(".");
+							}
+							if (combos.size() > thresh) {
+								Collections.sort(combos, comp);
+								combos = new ArrayList<Combo>(combos.subList(0,
+										results));
+							}
+						}
+					}
+				}
+			}
+		}
+		System.out.print("x");
+		return combos;
+
+	}
+
+	private List<Combo> callDef() throws Exception {
+		int total_def = 0;
+		int s_def = 0;
+		int p_def = 0;
+		int g_def = 0;
+		int a_def = 0;
+		int h_def = 0;
+
+		int i = 0;
+		for (Shoes s : shoes) {
+			s_p = s.getTotalPrice();
+			s_def = s.getTotaldef();
+			for (Pants p : pants) {
+				p_p = p.getTotalPrice();
+				p_def = p.getTotaldef();
+				for (Gloves g : gloves) {
+					g_p = g.getTotalPrice();
+					g_def = g.getTotaldef();
+					for (Armor a : armors) {
+						a_p = a.getTotalPrice();
+						a_def = a.getTotaldef();
+						for (Helm h : helms) {
+							h_p = h.getTotalPrice();
+							h_def = h.getTotaldef();
+							total_p = s_p + p_p + g_p + a_p + h_p;
+
+							total_def = s_def + p_def + h_def + a_def + g_def
+									+ h_def;
+							if (total_p <= PRICE
+									&& (total_def + MAXIS[0]) >= min) {
+								Combo c = new Combo(s, p, g, a, h);
+								if (c.getAtk() >= ATK && c.getMatk() >= MATK) {
+									combos.add(c);
+								}
+							}
+							i++;
+							if (i % 1000000 == 0) {
+								System.out.print(".");
+							}
+							if (combos.size() > thresh) {
+								Collections.sort(combos, comp);
+								combos = new ArrayList<Combo>(combos.subList(0,
+										results));
+								synchronized (this) {
+									min = combos.get(results - 1).getDef();
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+		System.out.print("x");
+		return combos;
+
+	}
+
+	private List<Combo> callAtk() throws Exception {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 11. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.